Builds hang if not in a git repository

### Checklist

- [x] I confirm there are no unresolved issues reported on the [Chocolatey Status page](https://status.chocolatey.org).
- [x] I have verified this is the correct repository for opening this issue.
- [x] I have verified no other issues exist related to my problem.
- [x] I have verified this is not an issue for a specific package.
- [x] I have verified this issue is not security related.
- [x] I confirm I am using **official**, and **not** unofficial, or modified, Chocolatey products.

### What You Are Seeing?

When running the build in a directory that is not a git repository, the build seems to hang with the last output being "Testing to see if valid git repository...".

### What is Expected?

If we're not in a git repository, it should determine that and take whatever action we deem appropriate (failure may be an option, or if we're only looking for a version number we could force one 🤔 )

### How Did You Get This To Happen?

Working on getting the Vagrantfile in `chocolatey/choco/tests` working on Hyper-V using the `chocolatey-test-environment` box, missed copying the `.git` directory, and the build would just hang.

### System Details

See the details of the `chocolatey-test-environment` vagrant box.

### Installed Packages

```bash
chocolatey|2.4.3
chocolatey-compatibility.extension|1.0.0
chocolatey-core.extension|1.4.0
chocolatey-dotnetfx.extension|1.0.1
chocolatey-visualstudio.extension|1.11.1
chocolatey-windowsupdate.extension|1.0.5
dotnetfx|4.8.0.20220524
git|2.48.1
git.install|2.48.1
KB2919355|1.0.20160915
KB2919442|1.0.20160915
KB2999226|1.0.20181019
KB3033929|1.0.5
KB3035131|1.0.3
netfx-4.8|4.8.0.20220524
netfx-4.8-devpack|4.8.0.20190930
Pester|5.7.1
vcredist140|14.42.34438.20250221
visualstudio2022community|117.13.3
visualstudio2022-workload-manageddesktop|1.0.2
visualstudio-installer|2.0.3
```

### Output Log

<details><summary>Log</summary>
<p>

```bash
PS C:\code\choco> ./build.bat --verbosity=diagnostic --target=CI --testExecutionType=none --shouldRunalyze=false --shouldRunNuGet=false|Tee-Object output
==================================================
============= WRAP POWERSHELL SCRIPT =============
==================================================
calling "C:\code\choco\build.ps1" with args --verbosity=diagnostic --target=CI --testExecutionType=none --shouldRunalyze=false --shouldRunNuGet=false
Preparing to run build script...
Running build script...
Skipping verification of assembly 'Cake.MyGet.Module, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.MyGet.Module, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.30.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.TeamCity.Module, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.TeamCity.Module, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.30.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.TFBuild.Module, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.TFBuild.Module, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.30.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.TravisCI.Module, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.TravisCI.Module, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.30.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.DotNetTool.Module, Version=0.4.0.0, Culture=neutral, PublicKeyToken=null'.
Registering module Cake.MyGet.Module.MyGetModule...
Registering module Cake.TeamCity.Module.TeamCityModule...
Registering module Cake.TFBuild.Module.TFBuildModule...
Registering module Cake.TravisCI.Module.TravisCIModule...
Registering module Cake.DotNetTool.Module.DotNetToolModule...
NuGet Config not specified. Will use NuGet default mechanism for resolving it.
Analyzing build script...
Analyzing C:/code/choco/recipe.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/addins.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/analyzing.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/build.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/buildData.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/buildProvider.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/configuration.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/configurationBuilder.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/credentials.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/dependencyCheck.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/docker.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/eazfuscator.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/environment.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/github-actions.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/gitreleasemanager.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/gitversion.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/ilmerge.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/localbuild.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/modules.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/notifcations.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/packages.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/packageSourceData.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/parameters.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/paths.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/psscriptanalyzer.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/sign.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/sonarqube.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/strongname.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/tasks.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/teamcity.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/testing.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/tools.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/toolsettings.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/transifex.cake...
Analyzing C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/version.cake...
Processing build script...
Installing tools...
Installing addins...
The addin Cake.Coverlet will reference Cake.Coverlet.dll.
The addin Cake.DependencyCheck will reference Cake.DependencyCheck.dll.
The addin Cake.Discord will reference Cake.Discord.dll.
The addin Cake.Docker will reference Cake.Docker.dll.
The addin Cake.Eazfuscator.Net will reference Cake.Eazfuscator.Net.dll.
The addin Cake.Figlet will reference Cake.Figlet.dll.
The addin Cake.FileHelpers will reference Cake.FileHelpers.dll.
The addin Cake.Git will reference Cake.Git.dll.
The addin Cake.Git will reference LibGit2Sharp.dll.
The addin Cake.Gulp will reference Cake.Gulp.dll.
The addin Cake.Incubator will reference Cake.Incubator.dll.
The addin Cake.Issues will reference Cake.Issues.dll.
The addin Cake.Issues.MsBuild will reference Cake.Issues.MsBuild.dll.
The addin Cake.Issues.MsBuild will reference Microsoft.Build.Framework.dll.
The addin Cake.Issues.MsBuild will reference Microsoft.Build.Utilities.Core.dll.
The addin Cake.Issues.MsBuild will reference StructuredLogger.dll.
The addin Cake.Issues.MsBuild will reference System.Collections.Immutable.dll.
The addin Cake.Issues.InspectCode will reference Cake.Issues.InspectCode.dll.
The addin Cake.Issues.Reporting will reference Cake.Issues.Reporting.dll.
The addin Cake.Issues.Reporting.Generic will reference Cake.Issues.Reporting.Generic.dll.
The addin Cake.Issues.Reporting.Generic will reference RazorEngine.dll.
The addin Cake.Issues.Reporting.Generic will reference System.Web.Razor.dll.
The addin Cake.Json will reference Cake.Json.dll.
The addin Cake.Kudu will reference Cake.Kudu.dll.
The addin Cake.Mastodon will reference Cake.Mastodon.dll.
The addin Cake.Npm will reference Cake.Npm.dll.
The addin Cake.PowerShell will reference Cake.Powershell.dll.
The addin Cake.PowerShell will reference System.Management.Automation.dll.
The addin Cake.ReSharperReports will reference Cake.ReSharperReports.dll.
The addin Cake.Slack will reference Cake.Slack.dll.
The addin Cake.Sonar will reference Cake.Sonar.dll.
The addin Cake.StrongNameSigner will reference Cake.StrongNameSigner.dll.
The addin Cake.StrongNameTool will reference cake.strongnametool.dll.
The addin Cake.Transifex will reference Cake.Transifex.dll.
The addin Cake.Twitter will reference Cake.Twitter.dll.
The addin MagicChunks will reference MagicChunks.Cake.dll.
The addin MagicChunks will reference MagicChunks.dll.
The addin Cake.Issues.PullRequests will reference Cake.Issues.PullRequests.dll.
The addin Cake.Issues.PullRequests.AppVeyor will reference Cake.Issues.PullRequests.AppVeyor.dll.
Skipping verification of assembly 'Cake.Coverlet, Version=2.5.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.DependencyCheck, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.DependencyCheck, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.22.0).
This assembly must reference at least Cake.Core version 0.26.0.
Another option is to downgrade Cake to an earlier version.
It's not recommended, but you can explicitly opt out of assembly verification
by configuring the Skip Verification setting to true
(i.e. command line parameter "--settings_skipverification=true",
environment variable "CAKE_SETTINGS_SKIPVERIFICATION=true",
read more about configuration at https://cakebuild.net/docs/fundamentals/configuration)
Skipping verification of assembly 'Cake.Discord, Version=0.2.1.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Docker, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Eazfuscator.Net, Version=0.1.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.Eazfuscator.Net, Version=0.1.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.28.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.Figlet, Version=1.2.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.Figlet, Version=1.2.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.28.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.FileHelpers, Version=3.2.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Git, Version=1.1.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'LibGit2Sharp, Version=0.27.0.0, Culture=neutral, PublicKeyToken=7cbde695407f0333'.
Skipping verification of assembly 'Cake.Gulp, Version=0.11.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.Gulp, Version=0.11.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.28.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.Incubator, Version=5.1.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Issues, Version=0.7.1.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Issues.MsBuild, Version=0.7.2.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Microsoft.Build.Framework, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
Skipping verification of assembly 'Microsoft.Build.Utilities.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
Skipping verification of assembly 'StructuredLogger, Version=2.0.94.0, Culture=neutral, PublicKeyToken=d4c7181801cb6448'.
Skipping verification of assembly 'System.Collections.Immutable, Version=1.2.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
Skipping verification of assembly 'Cake.Issues.InspectCode, Version=0.7.1.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Issues.Reporting, Version=0.7.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Issues.Reporting.Generic, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'RazorEngine, Version=3.10.0.0, Culture=neutral, PublicKeyToken=9ee697374c7e744a'.
Skipping verification of assembly 'System.Web.Razor, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Skipping verification of assembly 'Cake.Json, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Kudu, Version=0.8.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.Kudu, Version=0.8.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.28.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.Mastodon, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Npm, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.Npm, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.28.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.Powershell, Version=0.4.8.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.
Skipping verification of assembly 'Cake.ReSharperReports, Version=0.10.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'Cake.ReSharperReports, Version=0.10.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.28.0).
For best compatibility it should target Cake.Core version 0.33.0.
Skipping verification of assembly 'Cake.Slack, Version=0.13.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Sonar, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.StrongNameSigner, Version=0.1.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'cake.strongnametool, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'cake.strongnametool, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.25.0).
This assembly must reference at least Cake.Core version 0.26.0.
Another option is to downgrade Cake to an earlier version.
It's not recommended, but you can explicitly opt out of assembly verification
by configuring the Skip Verification setting to true
(i.e. command line parameter "--settings_skipverification=true",
environment variable "CAKE_SETTINGS_SKIPVERIFICATION=true",
read more about configuration at https://cakebuild.net/docs/fundamentals/configuration)
Skipping verification of assembly 'Cake.Transifex, Version=1.0.1.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Twitter, Version=0.11.1.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'MagicChunks.Cake, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
The assembly 'MagicChunks.Cake, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.23.0).
This assembly must reference at least Cake.Core version 0.26.0.
Another option is to downgrade Cake to an earlier version.
It's not recommended, but you can explicitly opt out of assembly verification
by configuring the Skip Verification setting to true
(i.e. command line parameter "--settings_skipverification=true",
environment variable "CAKE_SETTINGS_SKIPVERIFICATION=true",
read more about configuration at https://cakebuild.net/docs/fundamentals/configuration)
Skipping verification of assembly 'MagicChunks, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Issues.PullRequests, Version=0.7.0.0, Culture=neutral, PublicKeyToken=null'.
Skipping verification of assembly 'Cake.Issues.PullRequests.AppVeyor, Version=0.7.0.0, Culture=neutral, PublicKeyToken=null'.
Adding assembly reference to mscorlib.dll...
Adding assembly reference to System.Core.dll...
Adding assembly reference to Microsoft.CSharp.dll...
Adding assembly reference to Cake.Core.dll...
Adding assembly reference to Cake.Common.dll...
Adding assembly reference to Cake.exe...
Adding assembly reference to System.dll...
Adding assembly reference to System.Xml.dll...
Adding assembly reference to System.Xml.Linq.dll...
Adding assembly reference to System.Data.dll...
Adding assembly reference to System.Runtime.dll...
Adding assembly reference to System.Collections.dll...
Adding assembly reference to System.Net.Http.dll...
Adding assembly reference to netstandard.dll...
Adding assembly reference to Cake.Coverlet.dll...
Adding assembly reference to Cake.DependencyCheck.dll...
Adding assembly reference to Cake.Discord.dll...
Adding assembly reference to Cake.Docker.dll...
Adding assembly reference to Cake.Eazfuscator.Net.dll...
Adding assembly reference to Cake.Figlet.dll...
Adding assembly reference to Cake.FileHelpers.dll...
Adding assembly reference to Cake.Git.dll...
Adding assembly reference to LibGit2Sharp.dll...
Adding assembly reference to Cake.Gulp.dll...
Adding assembly reference to Cake.Incubator.dll...
Adding assembly reference to Cake.Issues.dll...
Adding assembly reference to Cake.Issues.MsBuild.dll...
Adding assembly reference to Microsoft.Build.Framework.dll...
Adding assembly reference to Microsoft.Build.Utilities.Core.dll...
Adding assembly reference to StructuredLogger.dll...
Adding assembly reference to System.Collections.Immutable.dll...
Adding assembly reference to Cake.Issues.InspectCode.dll...
Adding assembly reference to Cake.Issues.Reporting.dll...
Adding assembly reference to Cake.Issues.Reporting.Generic.dll...
Adding assembly reference to RazorEngine.dll...
Adding assembly reference to System.Web.Razor.dll...
Adding assembly reference to Cake.Json.dll...
Adding assembly reference to Cake.Kudu.dll...
Adding assembly reference to Cake.Mastodon.dll...
Adding assembly reference to Cake.Npm.dll...
Adding assembly reference to Cake.Powershell.dll...
Adding assembly reference to System.Management.Automation.dll...
Adding assembly reference to Cake.ReSharperReports.dll...
Adding assembly reference to Cake.Slack.dll...
Adding assembly reference to Cake.Sonar.dll...
Adding assembly reference to Cake.StrongNameSigner.dll...
Adding assembly reference to cake.strongnametool.dll...
Adding assembly reference to Cake.Transifex.dll...
Adding assembly reference to Cake.Twitter.dll...
Adding assembly reference to MagicChunks.Cake.dll...
Adding assembly reference to MagicChunks.dll...
Adding assembly reference to Cake.Issues.PullRequests.dll...
Adding assembly reference to Cake.Issues.PullRequests.AppVeyor.dll...
Importing namespace Cake.Common...
Importing namespace Cake.Common.Build...
Importing namespace Cake.Common.Build.AppVeyor...
Importing namespace Cake.Common.Build.AppVeyor.Data...
Importing namespace Cake.Common.Build.AzurePipelines...
Importing namespace Cake.Common.Build.AzurePipelines.Data...
Importing namespace Cake.Common.Build.Bamboo...
Importing namespace Cake.Common.Build.Bamboo.Data...
Importing namespace Cake.Common.Build.BitbucketPipelines...
Importing namespace Cake.Common.Build.BitbucketPipelines.Data...
Importing namespace Cake.Common.Build.Bitrise...
Importing namespace Cake.Common.Build.Bitrise.Data...
Importing namespace Cake.Common.Build.ContinuaCI...
Importing namespace Cake.Common.Build.ContinuaCI.Data...
Importing namespace Cake.Common.Build.GitHubActions...
Importing namespace Cake.Common.Build.GitHubActions.Data...
Importing namespace Cake.Common.Build.GitLabCI...
Importing namespace Cake.Common.Build.GitLabCI.Data...
Importing namespace Cake.Common.Build.GoCD...
Importing namespace Cake.Common.Build.GoCD.Data...
Importing namespace Cake.Common.Build.Jenkins...
Importing namespace Cake.Common.Build.Jenkins.Data...
Importing namespace Cake.Common.Build.MyGet...
Importing namespace Cake.Common.Build.TeamCity...
Importing namespace Cake.Common.Build.TFBuild...
Importing namespace Cake.Common.Build.TFBuild.Data...
Importing namespace Cake.Common.Build.TravisCI...
Importing namespace Cake.Common.Build.TravisCI.Data...
Importing namespace Cake.Common.Diagnostics...
Importing namespace Cake.Common.IO...
Importing namespace Cake.Common.IO.Paths...
Importing namespace Cake.Common.Net...
Importing namespace Cake.Common.Security...
Importing namespace Cake.Common.Solution...
Importing namespace Cake.Common.Solution.Project...
Importing namespace Cake.Common.Solution.Project.Properties...
Importing namespace Cake.Common.Solution.Project.XmlDoc...
Importing namespace Cake.Common.Text...
Importing namespace Cake.Common.Tools...
Importing namespace Cake.Common.Tools.Cake...
Importing namespace Cake.Common.Tools.Chocolatey...
Importing namespace Cake.Common.Tools.Chocolatey.ApiKey...
Importing namespace Cake.Common.Tools.Chocolatey.Config...
Importing namespace Cake.Common.Tools.Chocolatey.Download...
Importing namespace Cake.Common.Tools.Chocolatey.Features...
Importing namespace Cake.Common.Tools.Chocolatey.Install...
Importing namespace Cake.Common.Tools.Chocolatey.New...
Importing namespace Cake.Common.Tools.Chocolatey.Pack...
Importing namespace Cake.Common.Tools.Chocolatey.Pin...
Importing namespace Cake.Common.Tools.Chocolatey.Push...
Importing namespace Cake.Common.Tools.Chocolatey.Sources...
Importing namespace Cake.Common.Tools.Chocolatey.Uninstall...
Importing namespace Cake.Common.Tools.Chocolatey.Upgrade...
Importing namespace Cake.Common.Tools.DotCover...
Importing namespace Cake.Common.Tools.DotCover.Analyse...
Importing namespace Cake.Common.Tools.DotCover.Cover...
Importing namespace Cake.Common.Tools.DotCover.Merge...
Importing namespace Cake.Common.Tools.DotCover.Report...
Importing namespace Cake.Common.Tools.DotNetCore...
Importing namespace Cake.Common.Tools.DotNetCore.Build...
Importing namespace Cake.Common.Tools.DotNetCore.BuildServer...
Importing namespace Cake.Common.Tools.DotNetCore.Clean...
Importing namespace Cake.Common.Tools.DotNetCore.Execute...
Importing namespace Cake.Common.Tools.DotNetCore.MSBuild...
Importing namespace Cake.Common.Tools.DotNetCore.NuGet.Delete...
Importing namespace Cake.Common.Tools.DotNetCore.NuGet.Push...
Importing namespace Cake.Common.Tools.DotNetCore.NuGet.Source...
Importing namespace Cake.Common.Tools.DotNetCore.Pack...
Importing namespace Cake.Common.Tools.DotNetCore.Publish...
Importing namespace Cake.Common.Tools.DotNetCore.Restore...
Importing namespace Cake.Common.Tools.DotNetCore.Run...
Importing namespace Cake.Common.Tools.DotNetCore.Test...
Importing namespace Cake.Common.Tools.DotNetCore.Tool...
Importing namespace Cake.Common.Tools.DotNetCore.VSTest...
Importing namespace Cake.Common.Tools.DupFinder...
Importing namespace Cake.Common.Tools.Fixie...
Importing namespace Cake.Common.Tools.GitLink...
Importing namespace Cake.Common.Tools.GitReleaseManager...
Importing namespace Cake.Common.Tools.GitReleaseManager.AddAssets...
Importing namespace Cake.Common.Tools.GitReleaseManager.Close...
Importing namespace Cake.Common.Tools.GitReleaseManager.Create...
Importing namespace Cake.Common.Tools.GitReleaseManager.Discard...
Importing namespace Cake.Common.Tools.GitReleaseManager.Export...
Importing namespace Cake.Common.Tools.GitReleaseManager.Label...
Importing namespace Cake.Common.Tools.GitReleaseManager.Open...
Importing namespace Cake.Common.Tools.GitReleaseManager.Publish...
Importing namespace Cake.Common.Tools.GitReleaseNotes...
Importing namespace Cake.Common.Tools.GitVersion...
Importing namespace Cake.Common.Tools.ILMerge...
Importing namespace Cake.Common.Tools.ILRepack...
Importing namespace Cake.Common.Tools.InnoSetup...
Importing namespace Cake.Common.Tools.InspectCode...
Importing namespace Cake.Common.Tools.MSBuild...
Importing namespace Cake.Common.Tools.MSpec...
Importing namespace Cake.Common.Tools.MSTest...
Importing namespace Cake.Common.Tools.NSIS...
Importing namespace Cake.Common.Tools.NuGet...
Importing namespace Cake.Common.Tools.NuGet.Add...
Importing namespace Cake.Common.Tools.NuGet.Delete...
Importing namespace Cake.Common.Tools.NuGet.Init...
Importing namespace Cake.Common.Tools.NuGet.Install...
Importing namespace Cake.Common.Tools.NuGet.List...
Importing namespace Cake.Common.Tools.NuGet.Pack...
Importing namespace Cake.Common.Tools.NuGet.Push...
Importing namespace Cake.Common.Tools.NuGet.Restore...
Importing namespace Cake.Common.Tools.NuGet.SetApiKey...
Importing namespace Cake.Common.Tools.NuGet.SetProxy...
Importing namespace Cake.Common.Tools.NuGet.Sources...
Importing namespace Cake.Common.Tools.NuGet.Update...
Importing namespace Cake.Common.Tools.NUnit...
Importing namespace Cake.Common.Tools.OctopusDeploy...
Importing namespace Cake.Common.Tools.OpenCover...
Importing namespace Cake.Common.Tools.ReportGenerator...
Importing namespace Cake.Common.Tools.ReportUnit...
Importing namespace Cake.Common.Tools.Roundhouse...
Importing namespace Cake.Common.Tools.SignTool...
Importing namespace Cake.Common.Tools.SpecFlow...
Importing namespace Cake.Common.Tools.SpecFlow.StepDefinitionReport...
Importing namespace Cake.Common.Tools.SpecFlow.TestExecutionReport...
Importing namespace Cake.Common.Tools.TextTransform...
Importing namespace Cake.Common.Tools.VSTest...
Importing namespace Cake.Common.Tools.VSWhere...
Importing namespace Cake.Common.Tools.VSWhere.All...
Importing namespace Cake.Common.Tools.VSWhere.Latest...
Importing namespace Cake.Common.Tools.VSWhere.Legacy...
Importing namespace Cake.Common.Tools.VSWhere.Product...
Importing namespace Cake.Common.Tools.WiX...
Importing namespace Cake.Common.Tools.WiX.Heat...
Importing namespace Cake.Common.Tools.XBuild...
Importing namespace Cake.Common.Tools.XUnit...
Importing namespace Cake.Common.Xml...
Importing namespace Cake.Core...
Importing namespace Cake.Core.Diagnostics...
Importing namespace Cake.Core.IO...
Importing namespace Cake.Core.Scripting...
Importing namespace Cake.Coverlet...
Importing namespace Cake.DependencyCheck...
Importing namespace Cake.Discord...
Importing namespace Cake.Discord.Chat...
Importing namespace Cake.Docker...
Importing namespace Cake.Eazfuscator.Net...
Importing namespace Cake.Figlet...
Importing namespace Cake.FileHelpers...
Importing namespace Cake.Git...
Importing namespace Cake.Gulp...
Importing namespace Cake.Incubator.AssertExtensions...
Importing namespace Cake.Incubator.DotNetBuildExtensions...
Importing namespace Cake.Incubator.DotNetCoreTestExtensions...
Importing namespace Cake.Incubator.EnumerableExtensions...
Importing namespace Cake.Incubator.FileExtensions...
Importing namespace Cake.Incubator.FilePathExtensions...
Importing namespace Cake.Incubator.FileSystemExtensions...
Importing namespace Cake.Incubator.GlobbingExtensions...
Importing namespace Cake.Incubator.LoggingExtensions...
Importing namespace Cake.Incubator.Project...
Importing namespace Cake.Incubator.ProjectPathExtensions...
Importing namespace Cake.Incubator.SolutionParserExtensions...
Importing namespace Cake.Incubator.StringExtensions...
Importing namespace Cake.Issues...
Importing namespace Cake.Issues.InspectCode...
Importing namespace Cake.Issues.MsBuild...
Importing namespace Cake.Issues.PullRequests...
Importing namespace Cake.Issues.PullRequests.AppVeyor...
Importing namespace Cake.Issues.Reporting...
Importing namespace Cake.Issues.Reporting.Generic...
Importing namespace Cake.Json...
Importing namespace Cake.Kudu...
Importing namespace Cake.Kudu.KuduSync...
Importing namespace Cake.Kudu.Provider...
Importing namespace Cake.Mastodon...
Importing namespace Cake.Npm...
Importing namespace Cake.Npm.AddUser...
Importing namespace Cake.Npm.Ci...
Importing namespace Cake.Npm.Install...
Importing namespace Cake.Npm.Pack...
Importing namespace Cake.Npm.Prune...
Importing namespace Cake.Npm.Publish...
Importing namespace Cake.Npm.Rebuild...
Importing namespace Cake.Npm.Set...
Importing namespace Cake.Npm.Update...
Importing namespace Cake.Npm.Version...
Importing namespace Cake.Powershell...
Importing namespace Cake.ReSharperReports...
Importing namespace Cake.Slack...
Importing namespace Cake.Slack.Chat...
Importing namespace Cake.Sonar...
Importing namespace Cake.StrongNameSigner...
Importing namespace Cake.StrongNameTool...
Importing namespace Cake.Transifex...
Importing namespace Cake.Twitter...
Importing namespace MagicChunks...
Importing namespace MagicChunks.Cake...
Importing namespace MagicChunks.Core...
Importing namespace Newtonsoft.Json.Linq...
Importing namespace System...
Importing namespace System.Collections.Generic...
Importing namespace System.Diagnostics...
Importing namespace System.IO...
Importing namespace System.Linq...
Importing namespace System.Management.Automation...
Importing namespace System.Text...
Importing namespace System.Threading.Tasks...
Importing namespace System.Xml.Linq...
Importing namespace System.Xml.XPath...
Compiling build script...
(4430,12): warning CS1701: Assuming assembly reference 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' used by 'Cake.Json' matches identity 'Newtonsoft.Js
on, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' of 'Newtonsoft.Json', you may need to supply runtime policy
(4436,12): warning CS1701: Assuming assembly reference 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' used by 'Cake.Json' matches identity 'Newtonsoft.Js
on, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' of 'Newtonsoft.Json', you may need to supply runtime policy
C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/build.cake(236,13): warning CS0618: 'CustomProjectParserResult.OutputPath' is obsolete: 'Use OutputPaths instead for multi-targeting s
upport'
C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/build.cake(238,44): warning CS0618: 'CustomProjectParserResult.OutputPath' is obsolete: 'Use OutputPaths instead for multi-targeting s
upport'
C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/build.cake(377,36): warning CS0618: 'CustomProjectParserResult.OutputPath' is obsolete: 'Use OutputPaths instead for multi-targeting s
upport'
C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/build.cake(388,32): warning CS0618: 'CustomProjectParserResult.OutputPath' is obsolete: 'Use OutputPaths instead for multi-targeting s
upport'
C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/build.cake(396,32): warning CS0618: 'CustomProjectParserResult.OutputPath' is obsolete: 'Use OutputPaths instead for multi-targeting s
upport'
C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/build.cake(440,29): warning CS0618: 'CustomProjectParserResult.OutputPath' is obsolete: 'Use OutputPaths instead for multi-targeting s
upport'
C:/code/choco/tools/Chocolatey.Cake.Recipe.0.28.4/Content/build.cake(441,36): warning CS0618: 'CustomProjectParserResult.OutputPath' is obsolete: 'Use OutputPaths instead for multi-targeting s
upport'
Using Local Build Provider...
Testing to see if valid git repository...
```

</p>
</details> 

### Additional Context

I noticed while updating this that it is using version 0.28.4 of the Chocolatey.Cake.Recipe, this is due to the branch I'm working on being a little behind chocolatey/choco's develop branch. I do not expect this to change when bringing it up to date, but will do that and edit this note when I do.

As suspected, there is no change in this behaviour with 0.29.0.

--- a/Chocolatey.Cake.Recipe/Content/gitversion.cake
+++ b/Chocolatey.Cake.Recipe/Content/gitversion.cake
@@ -52,7 +52,7 @@ public class BuildVersion
             };
         }
 
-        context.Information("Testing to see if valid git repository...");
+        context.Information("Testing to see if valid git repository for GitVersion use...");
 
         var rootPath = BuildParameters.RootDirectoryPath;
         

--- a/Chocolatey.Cake.Recipe/Content/gitversion.cake
+++ b/Chocolatey.Cake.Recipe/Content/gitversion.cake
@@ -52,14 +52,15 @@ public class BuildVersion
             };
         }
 
-        try
-        {
-            context.Information("Testing to see if valid git repository...");
+        context.Information("Testing to see if valid git repository...");
 
-            var rootPath = BuildParameters.RootDirectoryPath;
+        var rootPath = BuildParameters.RootDirectoryPath;
+        
+        if (context.GitIsValidRepository(rootPath))
+        {
             rootPath = context.GitFindRootFromPath(rootPath);
         }
-        catch (LibGit2Sharp.RepositoryNotFoundException)
+        else
         {
             context.Warning("Unable to locate git repository, so GitVersion can't be executed, returning default version numbers...");
 

--- a/Chocolatey.Cake.Recipe/Content/localbuild.cake
+++ b/Chocolatey.Cake.Recipe/Content/localbuild.cake
@@ -68,7 +68,7 @@ public class LocalBuildRepositoryInfo : IRepositoryInfo
 {
     public LocalBuildRepositoryInfo(ICakeContext context)
     {
-        context.Information("Testing to see if valid git repository...");
+        context.Information("Testing to see if valid git repository for LocalBuild setup...");
 
         if (context.GitIsValidRepository(context.MakeAbsolute(context.Environment.WorkingDirectory)))
         {

--- a/Chocolatey.Cake.Recipe/Content/localbuild.cake
+++ b/Chocolatey.Cake.Recipe/Content/localbuild.cake
@@ -68,9 +68,10 @@ public class LocalBuildRepositoryInfo : IRepositoryInfo
 {
     public LocalBuildRepositoryInfo(ICakeContext context)
     {
-        try
+        context.Information("Testing to see if valid git repository...");
+
+        if (context.GitIsValidRepository(context.MakeAbsolute(context.Environment.WorkingDirectory)))
         {
-            context.Information("Testing to see if valid git repository...");
 
             // Normally, would use BuildParameters.RootDirectoryPath here, but since
             // BuildProvider is executed before the Setup Task has executed, this property
@@ -143,7 +144,7 @@ public class LocalBuildRepositoryInfo : IRepositoryInfo
                 Tag = new LocalBuildTagInfo(context);
             }
         }
-        catch (LibGit2Sharp.RepositoryNotFoundException)
+        else
         {
             context.Warning("Unable to locate git repository, setting default values for repository properties...");
 

